std-bits/table/src/main/java/org/enso/table/data/column/builder/DoubleBuilder.java

```diff
@@ -22,44 +20,35 @@
 import org.enso.table.util.BitSets;
 
 /** A builder for floating point columns. */
-public class DoubleBuilder extends NumericBuilder {
-  DoubleBuilder(
-      BitSet isNothing, long[] data, int currentSize, ProblemAggregator problemAggregator) {
-    super(isNothing, data, currentSize);
+public class DoubleBuilder extends NumericBuilder implements BuilderForDouble {
+  protected final PrecisionLossAggregator precisionLossAggregator;
+  protected double[] data;
+
+  DoubleBuilder(int initialSize, ProblemAggregator problemAggregator) {
+    super();
+    this.data = new double[initialSize];
     precisionLossAggregator = new PrecisionLossAggregator(problemAggregator);
   }
 
   @Override
-  public void retypeToMixed(Object[] items) {
-    throw new IllegalStateException(
-        "The DoubleBuilder cannot be retyped to the Mixed type, because it would lose type"
-            + " information about integers that were converted to doubles. If recasting is needed,"
-            + " InferringDoubleBuilder should be used instead. This error leaking is a bug in the"
-            + " Table library.");
+  protected int getDataSize() {
+    return data.length;
   }
 
   @Override
-  public boolean canRetypeTo(StorageType type) {
-    return type instanceof BigDecimalType;
+  protected void resize(int desiredCapacity) {
+    double[] newData = new double[desiredCapacity];
+    System.arraycopy(data, 0, newData, 0, currentSize);
+    data = newData;
   }
 
   @Override
-  public TypedBuilder retypeTo(StorageType type) {
-    if (type instanceof BigDecimalType) {
-      BigDecimalBuilder res = new BigDecimalBuilder(currentSize);
-      for (int i = 0; i < currentSize; i++) {
-        if (isNothing.get(i)) {
-          res.appendNulls(1);
-        } else {
-          double d = Double.longBitsToDouble(data[i]);
-          BigDecimal bigDecimal = BigDecimal.valueOf(d);
-          res.appendNoGrow(bigDecimal);
-        }
-      }
-      return res;
-    } else {
-      throw new UnsupportedOperationException();
-    }
+  public void copyDataTo(Object[] items) {
+    throw new UnsupportedOperationException(
+        "The DoubleBuilder cannot be copied to Object[], because it would lose type"
```

Comment:
I love this clear error that would catch a bug and clearly tells us how to fix it if we get it <3